Sources/SwiftCliCore/InputValidator.swift

```diff
@@ -29,21 +45,35 @@ public class InputValidator {
                 throw error
             }
         }
+        return pattern
+    }
 
-        let expandedNestedArray = patternNestedArray.map { NestedArrayBuilder().expandRow(row: $0) }
-
-        let patternMetaData = MetaDataBuilder().buildAllMetaData(stitchArray: expandedNestedArray)
-
-        let isPatternMathematicallySound = validateEachRowWidth(allRowsMetaData: patternMetaData)
+    private func checkNoMathmaticalIssuesInArrayOfRowInfo(pattern: [RowInfo]) throws -> [RowInfo] {
+        let isPatternMathematicallySound = validateEachRowWidth(allRowsMetaData: pattern)
         switch isPatternMathematicallySound {
         case .success:
-            return patternMetaData
+            return pattern
         case .failure(let error):
             throw error
         }
     }
 
 
+
+    private func check(pattern: [[String]]) throws -> [[String]] {
```

Comment:
Is this a duplicate of `checkNoInvalidStitchesInNestedArray()`?

Reply:
Oops! Thank you, deleted.